src/main/java/org/openstreetmap/atlas/utilities/conversion/ProtoTagListConverter.java

```diff
@@ -0,0 +1,89 @@
+package org.openstreetmap.atlas.utilities.conversion;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.proto.ProtoTag;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Converts back and forth between a List of ProtoTags and an OSM tag Map.
+ *
+ * @author lcram
+ */
+public class ProtoTagListConverter implements TwoWayConverter<List<ProtoTag>, Map<String, String>>
+{
+    private static final Logger logger = LoggerFactory.getLogger(ProtoTagListConverter.class);
+
+    private static final String TAG_DELIMITER = "=";
+
+    @Override
+    public List<ProtoTag> backwardConvert(final Map<String, String> osmTagMap)
+    {
+        final List<ProtoTag> protoTags = new ArrayList<>();
+        for (final Map.Entry<String, String> entry : osmTagMap.entrySet())
+        {
+            final ProtoTag.Builder tagBuilder = ProtoTag.newBuilder();
+            final String keyText;
+            final String valueText;
+
+            if (entry.getKey() == null)
+            {
+                logger.warn("buildProtoTagListFromTagMap found null key");
```

Comment:
Not sure if it is possible, but adding more context to those logs might help...

Reply:
It's definitely possible. I'll change that in the next couple of commits.